nexus/db-queries/benches/harness/mod.rs

```diff
@@ -0,0 +1,202 @@
+            ),
+          crdb_internal.statement_statistics ss
+      WHERE
+        blocking_txn_fingerprint_id = encode(ss.transaction_fingerprint_id, 'hex')) hce,
+      crdb_internal.statement_statistics ss2
+WHERE
+  hce.blocking_txn_fingerprint_id != '0000000000000000' AND
+  hce.waiting_txn_fingerprint_id != '0000000000000000' AND
+  hce.waiting_txn_fingerprint_id = encode(ss2.transaction_fingerprint_id, 'hex')
+ORDER BY
+  contention_count
+DESC;",
+        description: "Transaction statements which are blocking other statements",
+    }
+];
+
+impl TestHarness {
+    pub async fn new(log: &Logger, sled_count: usize) -> Self {
+        let db = TestDatabase::new_with_datastore(log).await;
+        let (opctx, datastore) = (db.opctx(), db.datastore());
+        let (authz_project, _project) =
+            create_project(&opctx, &datastore).await;
+        create_sleds(&datastore, sled_count).await;
+
+        Self { db, authz_project }
+    }
+
+    /// Emit internal CockroachDb information about contention
+    pub async fn print_contention(&self) {
+        let client =
+            self.db.crdb().connect().await.expect("Failed to connect to db");
+
+        // Used for padding: get a map of "column name" -> "max value length".
+        let max_lengths_by_column = |rows: &Vec<Row>| {
+            let mut lengths = HashMap::new();
+            for row in rows {
+                for column in &row.values {
+                    let value_len = column.value().unwrap().as_str().len();
+                    let name_len = column.name().len();
+                    let len = std::cmp::max(value_len, name_len);
+
+                    lengths
+                        .entry(column.name().to_string())
+                        .and_modify(|entry| {
+                            if len > *entry {
+                                *entry = len;
+                            }
+                        })
+                        .or_insert(len);
+                }
+            }
+            lengths
+        };
+
+        for ContentionQuery { sql, description } in QUERIES {
+            let rows = client
+                .query(sql, &[])
+                .await
+                .expect("Failed to query contended tables");
+            let rows = process_rows(&rows);
+            if rows.is_empty() {
+                continue;
+            }
+
+            println!("{description}");
+            let max_lengths = max_lengths_by_column(&rows);
+            let mut header = true;
+
+            for row in rows {
+                if header {
+                    let mut total_len = 0;
+                    for column in &row.values {
+                        let width = max_lengths.get(column.name()).unwrap();
+                        print!(" {:width$} ", column.name());
+                        print!("|");
+                        total_len += width + 3;
+                    }
+                    println!("");
```

Comment:
huh, i thought clippy disliked printlns with empty strings in them, but perhaps i made that up...